### PR TITLE
feat: chip

### PR DESCRIPTION
**Pull request by sterapps**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/valentingavran/anglify/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/docs/src/app/app.module.ts [hunk 1]
```diff
@@ -127,6 +130,7 @@ import { TooltipPageComponent } from './pages/component-pages/tooltip-page/toolt
     BottomNavigationExampleModule,
     SnackbarModule,
     SnackbarExamplesModule,
+    ChipExamplesModule,
```
Comment (iCrawl) [inline apps/docs/src/app/app.module.ts:133]:
```suggestion
    SnackbarExamplesModule,
    ChipModule,
    ChipExamplesModule,
```

File: apps/docs/src/app/examples/chip/appearances/appearances.component.html [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<anglify-chip>Filled</anglify-chip>
+<anglify-chip appearance="outlined"> Outlined</anglify-chip>
```
Comment (iCrawl) [inline apps/docs/src/app/examples/chip/appearances/appearances.component.html:2]:
```suggestion
<anglify-chip appearance="outlined">Outlined</anglify-chip>
```

File: apps/docs/src/app/examples/chip/filter/filter.component.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<anglify-chip-group multiple>
+  <anglify-chip filter>
+    <ng-template slot="prepend"> <anglify-icon icon="mdi-home" left></anglify-icon> </ng-template>
```
Comment (iCrawl) [inline apps/docs/src/app/examples/chip/filter/filter.component.html:3]:
```suggestion
    <ng-template slot="prepend">
      <anglify-icon icon="mdi-home" left></anglify-icon>
    </ng-template>
```

File: apps/docs/src/app/examples/chip/filter/filter.component.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Filled
+  </anglify-chip>
+  <anglify-chip filter> Filled </anglify-chip>
+  <anglify-chip filter appearance="outlined">
+    <ng-template slot="prepend"> <anglify-icon icon="mdi-home" left></anglify-icon> </ng-template>
```
Comment (iCrawl) [inline apps/docs/src/app/examples/chip/filter/filter.component.html:8]:
```suggestion
    <ng-template slot="prepend">
      <anglify-icon icon="mdi-home" left></anglify-icon>
    </ng-template>
```

File: libs/anglify/src/modules/chip/_variables.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+$chip-border-radius: var(--anglify-chip-border-radius, 16px);
+$chip-height: var(--anglify-chip-height, 18px);
```
Comment (iCrawl) [inline libs/anglify/src/modules/chip/_variables.scss:2]: Per M2 spec, a chip height is 32px.

File: libs/anglify/src/modules/chip/chip-settings.token.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { InjectionToken } from '@angular/core';
+import type { ChipSettings } from './chip.interface';
+
+export const DEFAULT_CHIP_SETTINGS: Required<ChipSettings> = {
+  appearance: 'filled',
+  filter: false,
+  size: 'regular',
```
Comment (iCrawl) [inline libs/anglify/src/modules/chip/chip-settings.token.ts:7]: We shouldn't do different sizes, we also removed this for buttons.

File: libs/anglify/src/modules/chip/chip.interface.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { ComponentSize } from '../../utils/interfaces';
+
+export type ChipAppearance = 'filled' | 'outlined';
+
+export interface ChipSettings {
+  appearance?: ChipAppearance;
+  filter?: boolean;
+  size?: ComponentSize;
```
Comment (iCrawl) [inline libs/anglify/src/modules/chip/chip.interface.ts:8]: See above.

**Review by iCrawl — CHANGES_REQUESTED** [review]

File: libs/anglify/src/modules/chip/_variables.scss [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+$chip-background-color: var(--anglify-chip-background-color, var(--color-surface));
```
Comment (iCrawl) [inline libs/anglify/src/modules/chip/_variables.scss]:
```suggestion
$chip-height: var(--anglify-chip-height, 20px);
$chip-background-color: var(--anglify-chip-background-color, var(--color-surface-overlay));
```

**Comment by iCrawl** [comment]
There still seems to be some sort of conflict here.

**Review by iCrawl — APPROVED** [review]

**Review by valentingavran — CHANGES_REQUESTED** [review]
Please ensure the following things:

- Add `@HostBinding("tabindex") public tabindex = 0;` to the chip component, so that it can be focused.
- Make sure that the style `&:focus-visible {}` & `&:hover{}` is defined and that the state colors changes with it
- Extract additional CSS variables (like padding  for example)
- Remove the ChipGroup component and use ItemGroup instead in the examples

**Review by valentingavran — APPROVED** [review]

**Pull request merged**